controllers/controller.js

```diff
@@ -1,25 +1,35 @@
-const {fetchTopcis,fetchID} = require("../models/model");
+const { fetchTopcis, fetchID } = require("../models/model");
 const endPoints = require("../endpoints.json");
 const getTopics = (req, res, next) => {
   fetchTopcis()
     .then((topic) => {
-      res.status(200).send({topic:topic});
-   
+      res.status(200).send({ topic: topic });
     })
     .catch((err) => {
       next(err);
     });
 };
 
 const getApi = (req, res) => {
-
-  res.status(200).send({endPoints});
+  res.status(200).send({ endPoints });
 };
-const getArticleId = (req,res,next)=>{
-  const articleID = req.params
-  fetchID(articleID).then((article)=>{
+const getArticleId = (req, res, next) => {
+  const { article_id } = req.params;
 
-  })
-  
+  fetchID(article_id)
+    .then((article) => {
+     
+      res.status(200).send({ article });
+    })
+    .catch((err) => {
+      console.log(err,"error here");
+if(err.status && err.msg){
+  res.status(err.status).send(err.msg)
+}
+if(err.code ==='22P02'){
+  res.status(400).send("Article not valid")
```

Comment:
You should extract these out into a error handler, let next(err) take them to the error handler and put this code there. 

Otherwise, you will have to repeat this code on every controller when you will have very similar errors to deal with